presto-main/src/main/java/com/facebook/presto/sql/gen/JoinCompiler.java

```diff
@@ -733,10 +749,12 @@ public PagesHashStrategy createPagesHashStrategy(List<? extends List<Block>> cha
     {
         private final List<Type> types;
         private final List<Integer> joinChannels;
+        private final List<ComparisonExpressionType> comparisons;
 
-        private CacheKey(List<? extends Type> types, List<Integer> joinChannels)
+        private CacheKey(List<? extends Type> types, List<ComparisonExpressionType> comparisons, List<Integer> joinChannels)
         {
             this.types = ImmutableList.copyOf(requireNonNull(types, "types is null"));
+            this.comparisons = ImmutableList.copyOf(requireNonNull(comparisons, "comparisons is null"));
```

Comment:
also check arg that the two lists are the same length. We can get some weird hard to find bugs otherwise